File: app/src/main/java/ai/rotor/rotorvehicle/ui/blackbox/BlackboxViewModel.java
package ai.rotor.rotorvehicle.ui.blackbox;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class BlackboxViewModel extends ViewModel {

    private MutableLiveData<String> mText;

    public BlackboxViewModel() {
        mText = new MutableLiveData<>();
        mText.setValue("This is the blackbox fragment");
    }

    public LiveData<String> getText() {
        return mText;
    }
}

File: app/src/main/java/ai/rotor/rotorvehicle/ui/blackbox/BlackboxViewModel.kt
package ai.rotor.rotorvehicle.ui.blackbox

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class BlackboxViewModel : ViewModel() {

    private val mText: MutableLiveData<String>

    val text: LiveData<String>
        get() = mText

    init {
        mText = MutableLiveData()
        mText.value = "This is the blackbox fragment"
    }
}